src/main.rs

```diff
@@ -58,15 +68,161 @@ async fn init_maker(
     }
 }
 
+fn init_rate_updates(
+    update_interval_secs: u64,
+) -> (
+    Pin<Box<dyn Future<Output = comit::Never> + Send>>,
+    Receiver<anyhow::Result<MidMarketRate>>,
+) {
+    let (mut sender, receiver) =
+        futures::channel::mpsc::channel::<anyhow::Result<MidMarketRate>>(5);
```

Comment:
Changing it to `0` (using a constant) , we should immediately consume on all channels due to the loop; given the current design it cannot not happen that a sender sends something while the receiver has not received the previous message.

@thomaseizinger I think this should also be true for the execution channel, because I am cloning the `sender` upon spawning the execution. Like for the other channels there should not be back-pressure, thus I think I can set all of the channels buffers to `0`. Please correct me if I am mistaken.

Reply:
Actually on second thought I am not sure this makes much sense...

I had to tell clippy to allow an unnecessary clone for this:

https://github.com/coblox/nectar/blob/b879d738cb92aa8eb4560025a730adbc16e0e48c/src/main.rs#L222-L225

I doubt that this really has the effect I think it has. @thomaseizinger if you have time it would be cool to discuss this :)

Reply:
Note: After discussion with @thomaseizinger we moved the clone into the loop (clippy does not complain about it there) - we looked at the expanded `select!` code because this is weird behaviour (the compiler should have complained already, because the clone is expected!). 

Thomas will look into this further and might upstream a question with a simplified example :)